# REAL TIME COMMUNICATION SYSTEM POWERED BY AI FOR SPECIALLY ABLED 

# IBM WATSON STUDIO DEPLOYMENT CODE

# 1.INSTALLING THE KERAS ,INSTALLING THE TENSORFLOW

###### !pip install Keras==2.2.4 !pip install tensorflow==2.7

# 2.IMPORTING LIBRARIES TO BUILD MODEL.

In [ ]:
#library to train the model
import keras
import tensorflow


from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense,Convolution2D,MaxPooling2D, Flatten

# 3.IMPORTING LIBRARIES FOR IMAGE AUGMENTATION.

In [ ]:
#image augmentation
from tensorflow.keras.preprocessing.image import ImageDataGenerator
train_datagen=ImageDataGenerator(rescale=1./255,zoom_range=0.2,shear_range=0.2,horizontal_flip=True,vertical_flip=False)
test_datagen=ImageDataGenerator(rescale=1./255)

# 4.ADDING STREAMING_BODY_OBJECT FOR DATASET.ZIP

In [ ]:
import os, types
import pandas as pd
from botocore.client import Config
import ibm_boto3

def __iter__(self): return 0

# @hidden_cell
# The following code accesses a file in your IBM Cloud Object Storage. It includes your credentials.
# You might want to remove those credentials before you share the notebook.
cos_client = ibm_boto3.client(service_name='s3',
    ibm_api_key_id='IMzFuAWRpYPnwh2XocJvGqTbHiPAMNnnEcIBBt8bQRGq',
    ibm_auth_endpoint="https://iam.cloud.ibm.com/oidc/token",
    config=Config(signature_version='oauth'),
    endpoint_url='https://s3.private.us.cloud-object-storage.appdomain.cloud')

bucket = 'realtimecommunication-donotdelete-pr-fx3wrumk8qzbvv'
object_key = 'Dataset.zip'

streaming_body_7 = cos_client.get_object(Bucket=bucket, Key=object_key)['Body']

# Your data file was loaded into a botocore.response.StreamingBody object.
# Please read the documentation of ibm_boto3 and pandas to learn more about the possibilities to load the data.
# ibm_boto3 documentation: https://ibm.github.io/ibm-cos-sdk-python/
# pandas documentation: http://pandas.pydata.org/

In [1]:
pwd

'C:\\Users\\nivet\\Desktop'

# 5.UNZIPPING THE DATASET

In [2]:
from io import BytesIO
import zipfile
unzip=zipfile.ZipFile(BytesIO(streaming_body_6.read()),'r')
file_paths=unzip.namelist()
for path in file_paths:
    unzip.extract(path)

NameError: name 'streaming_body_6' is not defined

In [3]:
ls

 Volume in drive C is Windows 10
 Volume Serial Number is 98E7-6D07

 Directory of C:\Users\nivet\Desktop

19-11-2022  12.01 AM    <DIR>          .
19-11-2022  12.01 AM    <DIR>          ..
18-11-2022  11.55 PM    <DIR>          .ipynb_checkpoints
13-10-2022  02.06 PM             2,216 4045 - Chrome.lnk
18-11-2022  11.29 AM           283,727 Add_The _Flatten_Layer.ipynb
18-11-2022  12.58 PM           283,727 Add_The _Flatten_Layer-Copy1.ipynb
18-11-2022  10.39 AM           283,431 Add_The_Pooling_Layer.ipynb
16-11-2022  08.28 PM    <DIR>          anaconda
18-11-2022  01.30 PM           282,118 Apply_Image Data Generator_Functionality_To_Train_And_Test_Set.ipynb
18-11-2022  11.50 PM             2,037 Final_Spyder_Deploy_Code.ipynb
18-11-2022  12.03 PM           287,241 Fit_And_Save_The_Model.ipynb
18-11-2022  02.34 PM           289,254 Import_The_Packages_And_Load_The_Saved_Modeel.ipynb
18-11-2022  02.14 PM           289,408 Import_The_Packages_And_Load_The_Saved_Modeel-Copy1.ipynb
18-1

In [4]:
pwd

'C:\\Users\\nivet\\Desktop'

In [ ]:
#checking that the dataset is there are not
import osC:\\Us
filenamer = os.listdir('C:\\USers\\nivet\\Desktop\Dataset\training_set')

# 6.TRAINING AND TESTING IMAGES UNDER CLASSES

In [6]:
x_train=train_datagen.flow_from_directory("C:\\USers\\nivet\\Desktop\Dataset\training_set",target_size=(64,64),class_mode="categorical",batch_size=25)

NameError: name 'train_datagen' is not defined

In [7]:
x_test=test_datagen.flow_from_directory("C:\\USers\\nivet\\Desktop\Dataset\training_set",target_size=(64,64),
class_mode='categorical' , batch_size=25)

NameError: name 'test_datagen' is not defined

# 7.TOTAL CLASSES UNDER TRAINING AND TESTING.

In [ ]:
x_train.class_indices

In [ ]:
x_test.class_indices

In [ ]:
train_datagen=ImageDataGenerator(rescale=1./255,zoom_range=0.2,horizontal_flip=True,vertical_flip=False)

In [ ]:
test_datagen=ImageDataGenerator(rescale=1./255)

# 8.MODEL BUILDING USING CNN

In [ ]:
model=Sequential()

In [ ]:
model.add(Convolution2D(32,(3,3),input_shape=(64,64,3),activation='relu'))

In [ ]:
model.add(MaxPooling2D(pool_size=(2,2)))

In [ ]:
model.add(MaxPooling2D(pool_size=(2,2)))

In [ ]:
model.add(Flatten())

In [ ]:
model.summary()

###### Model: "sequential"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
=================================================================
 conv2d (Conv2D)             (None, 62, 62, 32)        896       
                                                                 
 max_pooling2d (MaxPooling2D  (None, 31, 31, 32)       0         
 )                                                               
                                                                 
 flatten (Flatten)           (None, 30752)             0         
                                                                 
=================================================================
Total params: 896
Trainable params: 896
Non-trainable params: 0
_________________________________________________________________

# 9.ADDING LAYERS FOR MODEL TRAINING.
HIDDEN LAYERS

In [ ]:
model.add(Dense(units = 300, activation='relu'))
#model.add(Dense(unit = 150,init = "uniform" activation='softmax'))

# OUTPUT LAYERS

In [ ]:
model.add(Dense(units = 9, activation='softmax'))

# 10.OPTIMIZING THE MODEL

In [ ]:
model.compile(loss='categorical_crossentropy',optimizer='adam',metrics=['accuracy'])

In [ ]:
len(x_train)

630

In [ ]:
len(x_test)

# 11.FITTING THE MODEL

In [ ]:
### model.fit_generator(x_train,steps_per_epoch=len(x_train),validation_data=x_test,validation_steps=len(x_test),epochs=10)
# Fitting the Model Generator
model.fit_generator(x_train,steps_per_epoch=630,epochs=10,validation_data=x_test,validation_steps=90)
#model.fit(x_train, epochs=100, verbose=1)

###### /tmp/wsuser/ipykernel_2521/1177640488.py:3: UserWarning: `Model.fit_generator` is deprecated and will be removed in a future version. Please use `Model.fit`, which supports generators.
  model.fit_generator(x_train,steps_per_epoch=630,epochs=10,validation_data=x_test,validation_steps=90)
Epoch 1/10
630/630 [==============================] - 70s 111ms/step - loss: 0.2427 - accuracy: 0.9357 - val_loss: 0.2130 - val_accuracy: 0.9756
Epoch 2/10
630/630 [==============================] - 70s 112ms/step - loss: 0.0314 - accuracy: 0.9905 - val_loss: 0.2702 - val_accuracy: 0.9778
Epoch 3/10
630/630 [==============================] - 71s 113ms/step - loss: 0.0158 - accuracy: 0.9952 - val_loss: 0.3915 - val_accuracy: 0.9596
Epoch 4/10
630/630 [==============================] - 71s 112ms/step - loss: 0.0094 - accuracy: 0.9969 - val_loss: 0.3320 - val_accuracy: 0.9747
Epoch 5/10
630/630 [==============================] - 70s 111ms/step - loss: 0.0115 - accuracy: 0.9957 - val_loss: 0.3552 - val_accuracy: 0.9760
Epoch 6/10
630/630 [==============================] - 71s 112ms/step - loss: 0.0066 - accuracy: 0.9978 - val_loss: 0.3470 - val_accuracy: 0.9756
Epoch 7/10
630/630 [==============================] - 69s 110ms/step - loss: 0.0094 - accuracy: 0.9973 - val_loss: 0.3686 - val_accuracy: 0.9711
Epoch 8/10
630/630 [==============================] - 69s 110ms/step - loss: 0.0127 - accuracy: 0.9960 - val_loss: 0.7356 - val_accuracy: 0.9751
Epoch 9/10
630/630 [==============================] - 69s 109ms/step - loss: 0.0048 - accuracy: 0.9987 - val_loss: 0.3163 - val_accuracy: 0.9773
Epoch 10/10
630/630 [==============================] - 69s 109ms/step - loss: 0.0047 - accuracy: 0.9988 - val_loss: 0.4326 - val_accuracy: 0.9764


# 12.SAVING THE MODEL

In [11]:
ls

 Volume in drive C is Windows 10
 Volume Serial Number is 98E7-6D07

 Directory of C:\Users\nivet\Desktop

19-11-2022  12.23 AM    <DIR>          .
19-11-2022  12.23 AM    <DIR>          ..
18-11-2022  11.55 PM    <DIR>          .ipynb_checkpoints
13-10-2022  02.06 PM             2,216 4045 - Chrome.lnk
18-11-2022  11.29 AM           283,727 Add_The _Flatten_Layer.ipynb
18-11-2022  12.58 PM           283,727 Add_The _Flatten_Layer-Copy1.ipynb
18-11-2022  10.39 AM           283,431 Add_The_Pooling_Layer.ipynb
16-11-2022  08.28 PM    <DIR>          anaconda
18-11-2022  01.30 PM           282,118 Apply_Image Data Generator_Functionality_To_Train_And_Test_Set.ipynb
18-11-2022  11.50 PM             2,037 Final_Spyder_Deploy_Code.ipynb
18-11-2022  12.03 PM           287,241 Fit_And_Save_The_Model.ipynb
18-11-2022  02.34 PM           289,254 Import_The_Packages_And_Load_The_Saved_Modeel.ipynb
18-11-2022  02.14 PM           289,408 Import_The_Packages_And_Load_The_Saved_Modeel-Copy1.ipynb
18-1

In [12]:
pwd

'C:\\Users\\nivet\\Desktop'

In [ ]:
model.save('Dataset.h5')

In [13]:
Dataset.h5

NameError: name 'Dataset' is not defined

In [14]:
ls

 Volume in drive C is Windows 10
 Volume Serial Number is 98E7-6D07

 Directory of C:\Users\nivet\Desktop

19-11-2022  12.25 AM    <DIR>          .
19-11-2022  12.25 AM    <DIR>          ..
18-11-2022  11.55 PM    <DIR>          .ipynb_checkpoints
13-10-2022  02.06 PM             2,216 4045 - Chrome.lnk
18-11-2022  11.29 AM           283,727 Add_The _Flatten_Layer.ipynb
18-11-2022  12.58 PM           283,727 Add_The _Flatten_Layer-Copy1.ipynb
18-11-2022  10.39 AM           283,431 Add_The_Pooling_Layer.ipynb
16-11-2022  08.28 PM    <DIR>          anaconda
18-11-2022  01.30 PM           282,118 Apply_Image Data Generator_Functionality_To_Train_And_Test_Set.ipynb
18-11-2022  11.50 PM             2,037 Final_Spyder_Deploy_Code.ipynb
18-11-2022  12.03 PM           287,241 Fit_And_Save_The_Model.ipynb
18-11-2022  02.34 PM           289,254 Import_The_Packages_And_Load_The_Saved_Modeel.ipynb
18-11-2022  02.14 PM           289,408 Import_The_Packages_And_Load_The_Saved_Modeel-Copy1.ipynb
18-1

# 14.INSTALLING WATSON MACHINE LEARNING CLIENT SOFTWARE

In [16]:
#installing the machine learning repository
!pip install watson_machine_learning_client --upgrade

# 15.IMPORTING APICLIENT FOR DEPLOYING.

In [ ]:
from ibm_watson_machine_learning import APIClient
url_credentials = {
    "url": "https://us-south.ml.cloud.ibm.com",
    #"apikey": "sqLVTXSP3nnAKfzJ1rKRKCpNzS_XZ8_HXa9FRwV7BvOP"
    "apikey":  "yVlgJh_0MVtYQmrWl9PAa6M60YXRYSkm0BXYZjlfnmrz"
}
client = APIClient(url_credentials)

In [ ]:
client = APIClient(url_credentials)
client

# 16.CREATING API_CLIENT SPACE ID.

In [ ]:
def guid_from_space_name(client, space_name):
    space = client.spaces.get_details()
    return(next(item for item in space['resources'] if item['entity']['name'] == space_name)['metadata']['id'])

In [ ]:
space_uid = guid_from_space_name(client, 'newspace')
print("space UID = " + space_uid)

In [17]:
client.set.default_space(space_uid)

NameError: name 'client' is not defined

In [ ]:
client.software_specifications.list(500)

In [ ]:
software_spec_uid = client.software_specifications.get_uid_by_name("tensorflow_rt22.1-py3.9")
software_spec_uid

# 17.STORING THE MODEL_ID FOR DATASET.H5

In [ ]:
#store the model
model_details = client.repository.store_model(model='image-Classification-model_new.tgz',meta_props={
    client.repository.ModelMetaNames.NAME: "CNN",
    client.repository.ModelMetaNames.TYPE: "keras_2.2.4",
    client.repository.ModelMetaNames.SOFTWARE_SPEC_UID:software_spec_uid
})
model_id = client.repository.get_model_uid(model_details)

In [ ]:
model_details=client.repository.store_model(model="Dataset.tgz",meta_props={
client.repository.ModelMetaNames.NAME: "CNN Model Building",
client.repository.ModelMetaNames.TYPE: "tensorflow_2.7",
client.repository.ModelMetaNames.SOFTWARE_SPEC_UID: software_spec_uid
})

In [21]:
model_id

NameError: name 'model_id' is not defined

# 18.DOWNLOADING THE TAR FILE ON CLIENT REPOSITORY

In [ ]:
client.repository.download(model_id, 'my_model.tar.gz')

# 19.TEST THE MODEL

In [ ]:
import numpy as np
from tensorflow.keras.models import load_model
from keras.preprocessing import image

# 20.LOADING THE DATASET

In [ ]:
#Load the model
model=load_model('Dataset.h5')

# 21.ADDING STREAMING_BODY FOR TEST IMAGE.

# 22.TESTING ON SEVERAL TESTING IMAGES

In [ ]:
import os, types
import pandas as pd
from botocore.client import Config
import ibm_boto3

def __iter__(self): return 0

# @hidden_cell
# The following code accesses a file in your IBM Cloud Object Storage. It includes your credentials.
# You might want to remove those credentials before you share the notebook.
cos_client = ibm_boto3.client(service_name='s3',
    ibm_api_key_id='IMzFuAWRpYPnwh2XocJvGqTbHiPAMNnnEcIBBt8bQRGq',
    ibm_auth_endpoint="https://iam.cloud.ibm.com/oidc/token",
    config=Config(signature_version='oauth'),
    endpoint_url='https://s3.private.us.cloud-object-storage.appdomain.cloud')

bucket = 'realtimecommunication-donotdelete-pr-fx3wrumk8qzbvv'
object_key = '11.png'

streaming_body_8 = cos_client.get_object(Bucket=bucket, Key=object_key)['Body']

# Your data file was loaded into a botocore.response.StreamingBody object.
# Please read the documentation of ibm_boto3 and pandas to learn more about the possibilities to load the data.
# ibm_boto3 documentation: https://ibm.github.io/ibm-cos-sdk-python/
# pandas documentation: http://pandas.pydata.org/

In [22]:
ls

 Volume in drive C is Windows 10
 Volume Serial Number is 98E7-6D07

 Directory of C:\Users\nivet\Desktop

19-11-2022  12.39 AM    <DIR>          .
19-11-2022  12.39 AM    <DIR>          ..
18-11-2022  11.55 PM    <DIR>          .ipynb_checkpoints
13-10-2022  02.06 PM             2,216 4045 - Chrome.lnk
18-11-2022  11.29 AM           283,727 Add_The _Flatten_Layer.ipynb
18-11-2022  12.58 PM           283,727 Add_The _Flatten_Layer-Copy1.ipynb
18-11-2022  10.39 AM           283,431 Add_The_Pooling_Layer.ipynb
16-11-2022  08.28 PM    <DIR>          anaconda
18-11-2022  01.30 PM           282,118 Apply_Image Data Generator_Functionality_To_Train_And_Test_Set.ipynb
18-11-2022  11.50 PM             2,037 Final_Spyder_Deploy_Code.ipynb
18-11-2022  12.03 PM           287,241 Fit_And_Save_The_Model.ipynb
18-11-2022  02.34 PM           289,254 Import_The_Packages_And_Load_The_Saved_Modeel.ipynb
18-11-2022  02.14 PM           289,408 Import_The_Packages_And_Load_The_Saved_Modeel-Copy1.ipynb
18-1

In [23]:
pwd

'C:\\Users\\nivet\\Desktop'

In [24]:
cd \\'C:\\Users\\nivet\\Desktop'Dataset\test_set\

[WinError 53] The network path was not found: "\\\\'C:\\\\Users\\\\nivet\\\\Desktop'Dataset\\test_set\\"
C:\Users\nivet\Desktop


In [25]:
ls

 Volume in drive C is Windows 10
 Volume Serial Number is 98E7-6D07

 Directory of C:\Users\nivet\Desktop

19-11-2022  12.41 AM    <DIR>          .
19-11-2022  12.41 AM    <DIR>          ..
18-11-2022  11.55 PM    <DIR>          .ipynb_checkpoints
13-10-2022  02.06 PM             2,216 4045 - Chrome.lnk
18-11-2022  11.29 AM           283,727 Add_The _Flatten_Layer.ipynb
18-11-2022  12.58 PM           283,727 Add_The _Flatten_Layer-Copy1.ipynb
18-11-2022  10.39 AM           283,431 Add_The_Pooling_Layer.ipynb
16-11-2022  08.28 PM    <DIR>          anaconda
18-11-2022  01.30 PM           282,118 Apply_Image Data Generator_Functionality_To_Train_And_Test_Set.ipynb
18-11-2022  11.50 PM             2,037 Final_Spyder_Deploy_Code.ipynb
18-11-2022  12.03 PM           287,241 Fit_And_Save_The_Model.ipynb
18-11-2022  02.34 PM           289,254 Import_The_Packages_And_Load_The_Saved_Modeel.ipynb
18-11-2022  02.14 PM           289,408 Import_The_Packages_And_Load_The_Saved_Modeel-Copy1.ipynb
18-1

In [ ]:
ls

100.png  126.png  151.png  177.png  201.png  227.png  26.png  51.png  77.png
101.png  127.png  152.png  178.png  202.png  228.png  27.png  52.png  78.png
102.png  128.png  153.png  179.png  203.png  229.png  28.png  53.png  79.png
103.png  129.png  154.png  17.png   204.png  22.png   29.png  54.png  7.png
104.png  12.png   155.png  180.png  205.png  230.png  2.png   55.png  80.png
105.png  130.png  156.png  181.png  206.png  231.png  30.png  56.png  81.png
106.png  131.png  157.png  182.png  207.png  232.png  31.png  57.png  82.png
107.png  132.png  158.png  183.png  208.png  233.png  32.png  58.png  83.png
108.png  133.png  159.png  184.png  209.png  234.png  33.png  59.png  84.png
109.png  134.png  15.png   185.png  20.png   235.png  34.png  5.png   85.png
10.png   135.png  160.png  186.png  210.png  236.png  35.png  60.png  86.png
110.png  136.png  161.png  187.png  211.png  237.png  36.png  61.png  87.png
111.png  137.png  162.png  188.png  212.png  238.png  37.png  62.png  88.png
112.png  138.png  163.png  189.png  213.png  239.png  38.png  63.png  89.png
113.png  139.png  164.png  18.png   214.png  23.png   39.png  64.png  8.png
114.png  13.png   165.png  190.png  215.png  240.png  3.png   65.png  90.png
115.png  140.png  166.png  191.png  216.png  241.png  40.png  66.png  91.png
116.png  141.png  167.png  192.png  217.png  242.png  41.png  67.png  92.png
117.png  142.png  168.png  193.png  218.png  243.png  42.png  68.png  93.png
118.png  143.png  169.png  194.png  219.png  244.png  43.png  69.png  94.png
119.png  144.png  16.png   195.png  21.png   245.png  44.png  6.png   95.png
11.png   145.png  170.png  196.png  220.png  246.png  45.png  70.png  96.png
120.png  146.png  171.png  197.png  221.png  247.png  46.png  71.png  97.png
121.png  147.png  172.png  198.png  222.png  248.png  47.png  72.png  98.png
122.png  148.png  173.png  199.png  223.png  249.png  48.png  73.png  99.png
123.png  149.png  174.png  19.png   224.png  24.png   49.png  74.png  9.png
124.png  14.png   175.png  1.png    225.png  250.png  4.png   75.png
125.png  150.png  176.png  200.png  226.png  25.png   50.png  76.png

In [ ]:
x

In [27]:
import tensorflow as tf
tf .__version__

Using TensorFlow backend.


'2.10.1'

# 23.IBM DEPLOYMENT

In [ ]:
!pip install watson-machine-learning-client  

In [ ]:
from ibm_watson_machine learning import APIClient
wml_credentials={
"url":"https://us-south.ml.cloud.ibm.com",
"apikey":"x91CJTUTrrIfLvrXsKf8yLyI1KHb3JV0Y7Qrwy1zilb2"
}
client=APIClient(wml_credentials)

# CLIENT

In [ ]:
def guid_space_name(client,animal_deploy):
space-client.spaces.get_details()
return(next(item for item in space[' resources'] if iten['entity']['name']= animal_deploy)["metadata']['id'])

In [ ]:
space_uid-guid_space_name(client,'animal_deploy")
print("Space UID "+space_uid)

In [ ]:
client.set.default_space(space_uid)

In [ ]:
client,software specifications.list(200)

In [ ]:
software_space_uid=client.software_specifications.get_uid_by_name('tensorflow_rt22.1-py3.9¹)

In [ ]:
software_space_uid